lib/mappers/tn_mapper.py

```diff
@@ -0,0 +1,183 @@
+from akara import logger
+from dplaingestion.utilities import iterify
+from dplaingestion.selector import exists, getprop
+from dplaingestion.mappers.mods_mapper import MODSMapper
+
+
+class TNMapper(MODSMapper):
+    def __init__(self, provider_data):
+        super(TNMapper, self).__init__(provider_data)
+
+    def map_multiple_fields(self):
+        super(MODSMapper, self).map_multiple_fields()
+        self.map_alternative()
+        self.map_edm_has_type()
+
+    def map_alternative(self):
+        prop = "/metadata/mods/titleInfo"
+        if exists(self.provider_data, prop):
+            for titleInfoDict in iterify(getprop(self.provider_data, prop)):
+                titleInfo = titleInfoDict['title']
+                if isinstance(titleInfo, dict) and titleInfo['type'] == "alternative" and "#text" in titleInfo:
+                    self.update_source_resource({"alternative": titleInfo["#text"]})
+
+    def map_is_part_of(self):
+        path = "/metadata/mods/relatedItem"
+        if exists(self.provider_data, path):
+            for relatedItem in getprop(self.provider_data, path):
+                if "type" in relatedItem and "displayLabel" in relatedItem \
+                        and relatedItem["type"] == "host" and relatedItem["displayLabel"] == "Project":
+                    title = relatedItem["titleInfo"]["title"]
+                    self.update_source_resource({"isPartOf": title})
```

Comment:
I'm confused by the assignment to `sourceResource.isPartOf` because I don't see that property documented in either of our schema documents (https://dp.la/info/wp-content/uploads/2013/04/DPLA-MAP-V3.1-2.pdf or https://dp.la/info/wp-content/uploads/2015/03/MAPv4.pdf). In both cases, `isPartOf` is associated only with `sourceResource.collection`.  I _do_ see other mappings in `ingestion` that call `self.update_source_resource({"isPartOf": SOMETHING})` but I don't understand what's going on there.

@guegueng do you know?